File: Baekjoon/Java/소수&팰린드롬.java
import java.io.*;

import static java.lang.Math.sqrt;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int ans = 0;

        for (int i = n; i <= 1000001; i++) {
            if (i == 1) {
                continue;
            }
            if (palindrome(i)){
                if (find(i)) {
                    ans = i;
                    break;
                }
            }
        }

        if (ans == 0) {
            ans = 1003001;
        }
        System.out.println(ans);
    }

    public static boolean find(int n) {
        for (int i = 2; i < sqrt(n)+1; i++) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }

    public static boolean palindrome(int n) {
        String N = String.valueOf(n);
        StringBuilder sb = new StringBuilder(N);
        String reverse = sb.reverse().toString();
        return N.equals(reverse);
    }


}

File: Baekjoon/Python/소수&팰린드롬.py
import sys, math
input = sys.stdin.readline

n = int(input().rstrip())

def find(n):
    for i in range(2,int(math.sqrt(n)+1)):
        if n % i == 0:
            return False
    return True

ans = 0

for i in range(n, 1000001): 
    if i == 1:
        continue
    if str(i) == str(i)[::-1]: 
        if find(i): 
            ans = i
            break

if ans == 0: 
    ans = 1003001

print(ans)
